### PR TITLE
Add editorconfig

### PR DESCRIPTION
**Pull request by elsupergomez**
Add `editorconfig` for .NET coding convention settings
See https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-code-style-settings-reference

### CODE REVIEW

**Comment by andresmoschini** [comment]
❤️

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+# Core EditorConfig Options   #
+###############################
+# All files
+[*]
+indent_style = space
+# Code files
+[*.{cs,csx,vb,vbx}]
+indent_size = 4
+insert_final_newline = true
+charset = utf-8-bom
```
Comment (andresmoschini) [inline .editorconfig]: In other projects we are using `utf-8` for `[*]`. In some languages, the _BOM mark_ is a problem, in _C#_ it is not, but it is neither necessary.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+###############################
+# .NET Coding Conventions     #
+###############################
+[*.{cs,vb}]
+# Organize usings
+dotnet_sort_system_directives_first = true
```
Comment (andresmoschini) [inline .editorconfig]: If I am not wrong, you are not sorting _using directives_ in this way.

Did you test if these editor config rules are really working?

Comment (elsupergomez) [reply .editorconfig]: I think is better convention over configuration, for that, I want use the `.editorconfig` from template, and adapt current style and format

Thanks!

Comment (andresmoschini) [reply .editorconfig]: Right, and I saw that you have applied the style change in the files. ❤️

**Pull request merged**